frontend/lib/Providers/CompareEndpointsModel.dart

```diff
@@ -0,0 +1,105 @@
+import 'package:adpv_frontend/Repository/AbstractEndpointRepository.dart';
+import 'package:adpv_frontend/Repository/RestClient.dart';
+import 'package:dio/dio.dart';
+import 'package:flutter/cupertino.dart';
+
+import '../Models/Endpoint.dart';
+import '../Models/EndpointSummary.dart';
+
+class EndpointModel extends ChangeNotifier {
+  Map<String, EndpointSummary> endpointSummaryMap = {};
+  Map<String, Endpoint> endpointsMap = {};
+  Map<String, bool> selectedChips = {};
+  List<String> commonFields = [];
+  List<String> selectedEndpoints = [];
+  AbstractEndpointRepository repository = RestClient(Dio());
+
+  EndpointModel();
+
+  void selectChips(String label, bool value) {
+    selectedChips[label] = value;
+    if (value) {
+      selectedChips[label] = true;
+    } else {
+      selectedChips[label] = false;
+    }
+    notifyListeners();
+  }
+
+  void updateCommonFields() {
+    if (selectedEndpoints.isNotEmpty){
+      Map<String, int> counter = {};
+      List fields = endpointsMap[selectedEndpoints[0]]!.data.dataList[0].keys.toList();
+      for(String s in fields){
+        for(Endpoint e in endpointsMap.values){
+          if(e.hasField(s)) {
+            counter[s] = counter[s] == null ? 1 : counter[s]! + 1;
+          }
+        }
+      }
+      var max =0;
+      counter.forEach((key, value) {
+        if(value>max){
+          max= value;
+        }
+      });
```

Comment:
To można zamienić na:
```dart
var maxValue = counter.values.reduce(max);
```
Źródło [tutaj](https://stackoverflow.com/questions/20490868/dart-list-min-max-value).

Reply:
najsss <3